### PR TITLE
Toolset update: VS 2022 17.1 Preview 1

### PR DESCRIPTION
**Pull request by StephanTLavavej**
* Toolset update: VS 2022 17.1 Preview 1.
  + PowerShell 7.2.0 is now available.
  + I'm not increasing the required `_MSC_VER` yet, because the MSVC-internal build is still starting with a checked-in compiler version of 19.30.
  + Contains Nov 9 Patch Tuesday (Win11 22000.318).
  + Permanently set the environment variable `VSCMD_SKIP_SENDTELEMETRY` to `1`.
    - This works around VSO-1416950 "Blocking:  Send-VsDevShellTelemetry : Could not load file or assembly Newtonsoft.Json". Thanks to @vitek-karas for mentioning this workaround in dotnet/runtime#60061.
    - It's permanent because the STL's CI is totally unrepresentative of customer usage, noted in a comment.
  + In `provision-image.ps1`, simplify how we get/set the `Path`.
  + Also print messages when running `PipInstall`.
* Remove workaround for VSO-1168723 "Implicit copy assignment incorrectly not constexpr when subobject has explicit constexpr copy constructor".
* Remove workaround for DevCom-1511903 "Standard Library Header Units: Bogus error C2440 when convertible_to constrains a constructor".
* Remove workarounds for DevCom-1331017 "cl does not make special member functions implicitly constexpr".
  + `subrange` in `<xutility>` is still affected by something (possibly an unfixed corner case, possibly a different bug that was hiding behind the first bug). I've filed #2326 to track this, and preserved the workaround in product code.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -4,10 +4,10 @@
 # Build STL targeting x86, x64, arm, arm64
 
 variables:
-  tmpDir: 'D:\Temp'
-  buildOutputLocation: 'D:\build'
+  tmpDir: 'C:\stl_temp'
+  buildOutputLocation: 'C:\stl_build'
```
Comment (CaseyCarter) [inline azure-pipelines.yml:8]: Any chance these are changing from SSD to spinning rust and slowing our builds?

**Review by CaseyCarter — APPROVED** [review]

File: azure-pipelines.yml (hunk 1 above)
Comment (StephanTLavavej) [reply azure-pipelines.yml:8]: My understanding is that the OS disk is an SSD, but I can't seem to find a source that confirms that.

Comment (CaseyCarter) [reply azure-pipelines.yml:8]: I spose we can examine build times to see if they're in the same ballpark.

**Comment by StephanTLavavej** [comment]
This VMSS doesn't appear to have gotten the memo that Halloween is over, and is cursed:

```
'cmake' is not recognized as an internal or external command,
operable program or batch file.
##[error]Cmd.exe exited with code '1'.
```

(As usual, I watched the VMSS setup script to verify that VS installation succeeded.)

I think I'll abandon the attempt to the v5 SKU, going back to the old v4 SKU and the `D:\` directories.

File: azure-devops/cmake-configure-build.yml [hunk 2]
```diff
@@ -27,6 +27,7 @@ steps:
     if exist "$(${{ parameters.buildOutputLocationVar }})" (
       rmdir /S /Q "$(${{ parameters.buildOutputLocationVar }})"
     )
+    set VSCMD_SKIP_SENDTELEMETRY=1
```
Comment (AlexGuteniev) [inline azure-devops/cmake-configure-build.yml:30]: Should we have a brief explanation of the issue being fixed here?

**Review by CaseyCarter — APPROVED** [review]

File: stl/inc/xstring [hunk 3]
```diff
@@ -2168,9 +2165,6 @@ public:
 
     using _Mybase::_Mybase;
 
-    // TRANSITION, DevCom-1331017
-    _CONSTEXPR20 _String_iterator& operator=(const _String_iterator&) noexcept = default;
```
Comment (miscco) [inline stl/inc/xstring:2172]: Does that mean we can finally finish `constexpr optional`?

File: stl/inc/yvals_core.h [hunk 4]
```diff
@@ -1333,7 +1333,7 @@
 #define __cpp_lib_array_constexpr 201803L // P0858R0 Constexpr Iterator Requirements
 #endif // _HAS_CXX17
 
-#if _HAS_CXX20 && (defined(__clang__) || defined(__EDG__)) // TRANSITION, DevCom-1331017
+#if _HAS_CXX20
```
Comment (miscco) [inline stl/inc/yvals_core.h:1336]: 😹

**Review by miscco — APPROVED** [review]

File: azure-devops/cmake-configure-build.yml (hunk 2 above)
Comment (StephanTLavavej) [reply azure-devops/cmake-configure-build.yml:30]: Sure. I need to push changes to pick up Patch Tuesday so I can add comments too.

(There's probably a way to "bake in" this environment variable in provision-image although it's a bit more complicated)

Comment (StephanTLavavej) [reply azure-devops/cmake-configure-build.yml:30]: Yeah, I can bake it into `provision-image.ps1`. Update coming soon...

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed, or if more work is required.

**Review by MahmoudGSaleh — APPROVED** [review]

**Pull request merged**